app/grandchallenge/reader_studies/forms.py

```diff
@@ -643,28 +647,55 @@ def __init__(self, *args, pk, interface, reader_study, user, **kwargs):
         qs = ComponentInterface.objects.exclude(
             slug__in=reader_study.values_for_interfaces.keys()
         )
+        widget_kwargs = {}
+        attrs = {}
+        if pk is None and selected_interface is not None:
+            widget = Select
+        else:
+            widget = autocomplete.ModelSelect2
+            attrs.update(
+                {
+                    "data-placeholder": "Search for an interface ...",
+                    "data-minimum-input-length": 3,
+                    "data-theme": settings.CRISPY_TEMPLATE_PACK,
+                    "data-html": True,
+                }
+            )
+            widget_kwargs[
+                "url"
+            ] = "components:component-interface-autocomplete"
+            widget_kwargs["forward"] = ["interface"]
+
         if pk is not None:
-            attrs = {
-                "hx-get": reverse_lazy(
-                    "reader-studies:display-set-interfaces-create",
-                    kwargs={"pk": pk, "slug": reader_study.slug},
-                ),
-                "hx-target": f"#ds-content-{pk}",
-            }
+            attrs.update(
+                {
+                    "hx-get": reverse_lazy(
+                        "reader-studies:display-set-interfaces-create",
+                        kwargs={"pk": pk, "slug": reader_study.slug},
+                    ),
+                    "hx-target": f"#ds-content-{pk}",
+                    "hx-trigger": "interfaceSelected",
+                }
+            )
         else:
-            attrs = {
-                "hx-get": reverse_lazy(
-                    "reader-studies:display-set-new-interfaces-create",
-                    kwargs={"slug": reader_study.slug},
-                ),
-                "hx-target": f"#form-{kwargs['auto_id'][:-3]}",
-                "hx-swap": "outerHTML",
-                "disabled": selected_interface is not None,
-            }
+            attrs.update(
+                {
+                    "hx-get": reverse_lazy(
+                        "reader-studies:display-set-new-interfaces-create",
+                        kwargs={"slug": reader_study.slug},
+                    ),
+                    "hx-target": f"#form-{kwargs['auto_id'][:-3]}",
+                    "hx-swap": "outerHTML",
+                    "hx-trigger": "interfaceSelected",
+                    "disabled": selected_interface is not None,
+                }
+            )
+        widget_kwargs["attrs"] = attrs
+
         self.fields["interface"] = ModelChoiceField(
             initial=selected_interface,
             queryset=qs,
-            widget=Select(attrs=attrs),
+            widget=widget(**widget_kwargs),
```

Comment:
I'm not sure why, but trying this out, it doesn't give my any options in the select on the update and add-single-ds views. It's just an empty dropdown (even though the queryset that is passed here contains a bunch of options). So something is going wrong, but I haven't been able to find the issue.

Reply:
Having a look now.

Reply:
This is now fixed. It broke with the last commit.

Reply:
Doesn't that reveal an implementation issue as that form does not produce those widgets?

Reply:
The issue is that the `DisplaySetInterfacesCreateForm` is only ever loaded within either the update or create form, but the media for `_possible_widgets` is only loaded for the view's form_class, which is are the `DisplaySetCreateForm` for the create view and the `DisplaySetUpdateForm` for the update view. So in a sense the possible widgets for the `DisplaySetInterfacesCreateForm` are also widgets for the both the create and the update form. I think it would be clearer to add them as `*DisplaySetInterfacesCreateForm._possible_widgets` though.

Reply:
Those are different forms, it’s kind of hacky to fix a problem in the view by trying to fix it in the form attributes.

Reply:
Alternatively, instead of using the `form_class` attribute on the view to determine which media to include, we add a new attribute `included_form_classes` which is a list of all forms that could possibly be loaded in the view. Does that sounds better?

Reply:
Yes, then you have the possible forms for the view which provide the possible widgets.